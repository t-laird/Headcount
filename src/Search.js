import React, { Component } from 'react';
import './Styles/Search.css';

class Search extends Component {
  constructor() {
    super();

    this.state = {
      value: ''
    }
  } 

  render() {
    return (
      <div className='Search'>
        <input 
          type='text' 
          // value={this.state.value} 
          onChange={(e) => {
            this.props.updateQuery(e.target.value);
          }}
          placeholder='Search'/>
          <i className="icon-search"></i>

          
      </div>
    )
  }
}

export default Search;