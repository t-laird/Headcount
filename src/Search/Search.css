@import url(../../fontello-e98b1dbc/css/fontello.css);
.Search {
  display: flex;
  justify-content: flex-end;
  align-items: center;
  margin: 0px auto;
  width: 80%;
  max-width: 750px;
  font-size: 35px; }
  .Search input {
    font-size: 35px;
    padding-left: 5px;
    height: 50px;
    width: 0px;
    border: none;
    box-sizing: border-box;
    cursor: default;
    transition-duration: 0.8s; }
  .Search i {
    cursor: pointer;
    width: 49px; }

.Search.expanded input {
  width: calc(100% - 49px);
  cursor: auto;
  border-bottom: 1px solid black; }
