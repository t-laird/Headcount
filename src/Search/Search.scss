@import '../../fontello-e98b1dbc/css/searchicon.css';

.Search {
  display: flex;
  justify-content: flex-end;
  align-items: center;  
  margin: 0px auto 30px;;
  width: 80%;
  max-width: 750px;
  font-size: 35px;

  input {
    font-size: 35px;
    padding-left: 5px;
    height: 50px;
    width: 0px;
    border: none;
    box-sizing: border-box;
    cursor: default;
    transition-duration: 0.8s;
  }
  i {
    cursor: pointer;
    width: 49px;
  }
}

.Search.expanded {
  input {
    width: calc(100% - 49px);
    cursor: auto;
    border-bottom: 1px solid black;
  }
}