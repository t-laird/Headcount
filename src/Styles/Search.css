@import url(../../fontello-e98b1dbc/css/searchicon.css);
