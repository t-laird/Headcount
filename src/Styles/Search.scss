@import '../../fontello-e98b1dbc/css/searchicon.css';

