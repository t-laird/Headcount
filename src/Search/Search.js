import React, { Component } from 'react';
import PropTypes from 'prop-types';
import './Search.css';

class Search extends Component {
  constructor() {
    super();

    this.state = {
      expanded: 'Search',
      value: ''
    };
  }

  expandSearch() {
    if (this.state.expanded === 'Search expanded') {
      this.setState({
        expanded: 'Search',
      });
    } else {
      this.setState({
        expanded: 'Search expanded'
      });
      this.searchInput.focus();
    }
  }

  render() {
    return (
      <div className={this.state.expanded}>
        <i 
          className="icon-search" 
          onClick={()=> {this.expandSearch();}}></i>
        <input 
          type='text'
          value={this.state.value}
          ref={(sInput) => { this.searchInput = sInput; }}    
          onChange={(e) => {
            this.setState( {value: e.target.value} );
            this.props.updateQuery(e.target.value);
          }}
          placeholder='Search'
          onBlur={() => {
            this.setState({
              expanded: 'Search',
              value: ''
            });
          }}
        />
      </div>
    );
  }
}

Search.propTypes = {
  updateQuery: PropTypes.func,
};

export default Search;