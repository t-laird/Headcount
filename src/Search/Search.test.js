import React from 'react';
import ReactDOM from 'react-dom';
import Search from './Search';
import { shallow, mount } from 'enzyme';

describe('Search test', () => {
  let mockFunc;
  let renderedSearch;
  beforeEach(() => {
    mockFunc = jest.fn();
    renderedSearch = shallow(<Search updateQuery={mockFunc}/>);
  });
  it('should render the correct components', () => {
    expect(renderedSearch.find('.Search').length).toEqual(1);
  });

  it('should match the Search snapshot', () => {
    expect(renderedSearch).toMatchSnapshot();
  });

  it('should be instantiated with the default class of Search without the expanded class', () => {
    expect(renderedSearch.state('expanded')).toEqual('Search');
    expect(renderedSearch.find('.Search').length).toEqual(1);
    expect(renderedSearch.find('.expanded').length).toEqual(0);
  });

  it('Should change the class on the container when the search icon is clicked', () => {
    renderedSearch = mount(<Search updateQuery={mockFunc}/>);
    const searchButton = renderedSearch.find('.icon-search');
    searchButton.simulate('click');
    const expectedClass = 'Search expanded';
    expect(renderedSearch.state('expanded')).toEqual(expectedClass);
    expect(renderedSearch.find('.expanded').length).toEqual(1);
    expect(renderedSearch.find('.Search').length).toEqual(1);
  });

  it('Should call the function passed through props on input', () => {
    renderedSearch = mount(<Search updateQuery={mockFunc} />);
    renderedSearch.find('input').simulate('change', { target: {value: 'a'}});
    expect(mockFunc.mock.calls.length).toEqual(1);
    renderedSearch.find('input').simulate('change', { target: {value: 'ab'}});
    expect(mockFunc.mock.calls.length).toEqual(2);
  });

  it('Should collapse the search on blur from the input field', () => {
    renderedSearch = mount(<Search updateQuery={mockFunc} />);

    const searchButton = renderedSearch.find('.icon-search');
    searchButton.simulate('click');
    const initialExpectedClass = 'Search expanded';
    const finalExpectedClass = 'Search';

    expect(renderedSearch.state('expanded')).toEqual(initialExpectedClass);
    expect(renderedSearch.find('.expanded').length).toEqual(1);
    renderedSearch.find('input').simulate('blur');
    expect(renderedSearch.state('expanded')).toEqual(finalExpectedClass);
    expect(renderedSearch.find('.expanded').length).toEqual(0);    
  });
});