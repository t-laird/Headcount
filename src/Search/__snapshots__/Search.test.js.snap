exports[`Search test should match the Search snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Search
    updateQuery={[Function]} />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <i
          className="icon-search"
          onClick={[Function]} />,
        <input
          onBlur={[Function]}
          onChange={[Function]}
          placeholder="Search"
          type="text" />,
      ],
      "className": "Search",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "icon-search",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": "i",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "onBlur": [Function],
          "onChange": [Function],
          "placeholder": "Search",
          "type": "text",
        },
        "ref": [Function],
        "rendered": null,
        "type": "input",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <i
            className="icon-search"
            onClick={[Function]} />,
          <input
            onBlur={[Function]}
            onChange={[Function]}
            placeholder="Search"
            type="text" />,
        ],
        "className": "Search",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "icon-search",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": "i",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "onBlur": [Function],
            "onChange": [Function],
            "placeholder": "Search",
            "type": "text",
          },
          "ref": [Function],
          "rendered": null,
          "type": "input",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;
